listener.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"log"
+	"net"
+
+	baps3 "github.com/UniversityRadioYork/baps3-go"
+)
+
+type Listener struct {
+	listener    net.Listener
+	connectorCh chan baps3.Message
+	clientComm  chan []byte
+	tok         *baps3.Tokeniser
+	clients     []*Client
+}
+
+func MakeListener(addr string, port string, connectorCh chan baps3.Message) (*Listener, error) {
+	netListener, err := net.Listen("tcp", addr+":"+port)
+	if err != nil {
+		return nil, err
+	}
+	listener := &Listener{
+		listener:    netListener,
+		tok:         baps3.NewTokeniser(),
+		clientComm:  make(chan []byte),
+		connectorCh: connectorCh,
+	}
+	return listener, nil
+}
+
+func (l *Listener) Broadcast(msg baps3.Message) {
+	// Probably should have \n at the end
+	for _, c := range l.clients {
+		data, err := msg.Pack()
+		if err != nil {
+			continue // TODO Handle
+		}
+		c.Outgoing <- data
+	}
+}
+
+func (l *Listener) ReceiveLoop() {
+	for data := range l.clientComm {
+		lines, _, err := l.tok.Tokenise(data)
```

Comment:
Clients should not share a tokeniser really, as multiple partial requests will interfere with each other. This probably doesn't show itself as the client's reader blocks until `\n` but this is still not ideal. Every client should have it's own tokeniser and only send complete baps3.Message structs to the listener.